R/render.R

```diff
@@ -325,6 +330,13 @@ render <- function(input,
       rmarkdown.runtime = runtime
     )
 
+    # read root directory from argument (has precedence) or front matter
+    root_dir <- knit_root_dir
```

Comment:
Okay, so it works as I'm hoping it would already. One of the most common requests I've heard would be `knit_root_dir: ..` (i.e. my report is in a subdirectory of the "main" directory of my project from whence all paths should be resolved).